server/src/main/java/org/elasticsearch/search/fetch/FetchContext.java

```diff
@@ -167,6 +167,14 @@ public SearchHighlightContext highlight() {
         return searchContext.highlight();
     }
 
+    /**
+     * Does the index analyzer for this field have token filters that may produce
+     * backwards offsets in term vectors
+     */
+    public boolean containsBrokenAnalysis(String field) {
+        return mapperService().containsBrokenAnalysis(field);
```

Comment:
this also calls mapperService() :) can you add the method to QueryShardContext and call it from there?